### PR TITLE
Add create image by url

### PR DESCRIPTION
**Pull request by pzmrzy**
Fix Issue (#594 )

### CODE REVIEW

File: pkg/dispatchcli/cmd/create.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ import (
 	"github.com/vmware/dispatch/pkg/api/v1"
 	"github.com/vmware/dispatch/pkg/dispatchcli/i18n"
 	"github.com/vmware/dispatch/pkg/utils"
+	"os"
```
Comment (neosab) [inline pkg/dispatchcli/cmd/create.go]: Std Library imports go to the first block

Some pointers : https://github.com/golang/go/wiki/CodeReviewComments#imports

File: pkg/dispatchcli/cmd/create.go [hunk 2]
```diff
@@ -31,6 +33,7 @@ var (
 	createExample = i18n.T(``)
 	file          = i18n.T(``)
 	workDir       = i18n.T(``)
+	tmpPath      = i18n.T(`/tmp/seed.yaml`)
```
Comment (neosab) [inline pkg/dispatchcli/cmd/create.go]: Don't hardcode the tmpPath, use `ioutil.TempFile` and defer its removal.

File: pkg/dispatchcli/cmd/create.go [hunk 3]
```diff
@@ -63,6 +66,39 @@ func importFile(out io.Writer, errOut io.Writer, cmd *cobra.Command, args []stri
 	return importBytes(out, b, actionMap, actionName)
 }
 
+func importFileWithURL(out io.Writer, errOut io.Writer, cmd *cobra.Command, args []string, actionMap map[string]ModelAction, actionName string) error {
+	err := downloadFile(file, tmpPath)
```
Comment (neosab) [inline pkg/dispatchcli/cmd/create.go]: Actually, you don't even have to download the file to tmp dir, you could do this in-memory by reading the response and passing to `importBytes`

File: pkg/dispatchcli/cmd/create.go [hunk 4]
```diff
@@ -9,16 +9,16 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"github.com/ghodss/yaml"
```
Comment (neosab) [inline pkg/dispatchcli/cmd/create.go]: This is not stdlib, you should move it back.

File: pkg/dispatchcli/cmd/create.go [hunk 5]
```diff
@@ -9,16 +9,16 @@ import (
...
+	"github.com/pkg/errors"
```
Comment (neosab) [inline pkg/dispatchcli/cmd/create.go]: same

File: pkg/dispatchcli/cmd/create.go [hunk 6]
```diff
@@ -9,16 +9,16 @@ import (
...
+	"github.com/spf13/cobra"
```
Comment (neosab) [inline pkg/dispatchcli/cmd/create.go]: same here

**Comment by berndtj** [comment]
Is there an issue linked to this?  Please squash commits and add helpful message.

**Comment by berndtj** [comment]
Also, can we add an e2e for this?

**Comment by neosab** [comment]
Another improvement could be the resolution of Function `sourcePath`  which currently looks at the local filesystem. It will be nice if the CLI could try the local path else do a relative URL path while operating in this mode. I just realized the `seed.yaml` still won't work after this addition.

**Comment by neosab** [comment]
@pzmrzy Please add an E2E test for this. The test should use the `seed.yaml` that's part of the PR being tested by the CI job, otherwise, the `yaml` being tested may not be in sync with the API changes in that PR. This could lead to unnecessary CI failures.

**Pull request merged**